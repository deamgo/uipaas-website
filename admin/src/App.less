.siderbar {
  position: fixed;
  left: 0;
  top: 0;
  width: 143px;
  height: 100%;
  background-color: #FAFAFA;

  border: 1px solid #BFBFBF;

  .logo-wapper {
    position: absolute;
    left: 19px;
    top: 21px;

    img {
      width: 104px;
      height: 23px;
    }
  }

  .guide {
    position: absolute;
    left: 1px;
    top: 64px;
    width: 141px;
    height: 40px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #D4E9FD;
  }
}

.head {
  position: fixed;
  left: 143px;
  top: 0;
  width: 1150px;
  height: 64px;
  // background-color: skyblue;
  z-index: 1001;

  .cicle {
    position: absolute;
    right: 92px;
    top: 16px;
    width: 36px;
    height: 36px;
    background-color: #FAFAFA;
    border: 1px solid #BFBFBF;
    border-radius: 50%;
  }

  span {
    position: absolute;
    right: 34px;
    top: 20px;

    width: 48px;
    height: 32px;
  }
}

.main {
  position: absolute;
  left: 143px;
  top: 64px;
  padding: 0 42px;
  width: 1000px;
  height: auto;
  // background-color: lightblue;

  overflow: scroll;
}