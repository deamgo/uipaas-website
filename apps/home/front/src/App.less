* {
  padding: 0;
  margin: 0;
}

a {
  text-decoration: none;
}

ul {
  list-style: none;
}

body {
  width: 1920rem;
  height: 2501rem;
}

header {
  // position: fixed;
  position: absolute;
  left: 0;
  top: 0;
  width: 1920rem;
  z-index: 1001;
  translate: .5;

  .head {
    width: 100%;
    height: 120rem;
    background-color: #FFFFFF;

    box-sizing: border-box;

    .logo {
      position: absolute;
      left: 240rem;
      top: 50rem;
      width: 181rem;
      height: 26rem;
    }

    .reserveButton {

      position: absolute;
      right: 240rem;
      top: 30rem;
      width: 160rem;
      height: 60rem;
      border-radius: 8rem;
      border: 0;
      opacity: 1;

      background: #0871F0;

      cursor: pointer;

      span {
        /* Book demo */

        position: absolute;
        left: 34rem;
        top: 18rem;
        width: 93rem;
        height: 24rem;
        opacity: 1;

        font-family: Poppins;
        font-size: 16rem;
        font-weight: bold;
        line-height: normal;
        text-align: center;
        letter-spacing: 0em;

        color: #FFFFFF;
      }
    }
  }

  .head-fixed-top {
    border-width: 0 0 1px 0;
    border-style: solid;
    border-color: #E2ECF9;
    // opacity: 0.3;
    box-shadow: 0px 3px 8px 0px rgba(142, 182, 231, 0.1);
  }
}

main {
  width: 1920rem;

  .container {
    position: relative;
    height: 2501rem;

    .title {
      position: absolute;
      left: 240rem;
      top: 244rem;
      width: 887rem;
      height: 240rem;
      opacity: 1;

      font-family: Poppins;
      font-size: 64rem;
      font-weight: bold;
      line-height: 80rem;
      letter-spacing: 0em;

      color: #333333;

    }

    .dsp {
      position: absolute;
      left: 240rem;
      top: 508rem;
      width: 690rem;
      height: 60rem;
      opacity: 1;

      font-family: Poppins;
      font-size: 20rem;
      font-weight: 500;
      line-height: normal;
      letter-spacing: 0em;

      color: #5A5A5A;

    }

    .reserveButton {

      position: absolute;
      left: 240rem;
      top: 628rem;
      width: 160rem;
      height: 60rem;
      border-radius: 8rem;
      border: 0;
      opacity: 1;

      background: #0871F0;

      cursor: pointer;

      span {
        /* Book demo */

        position: absolute;
        left: 34rem;
        top: 18rem;
        width: 93rem;
        height: 24rem;
        opacity: 1;

        font-family: Poppins;
        font-size: 16rem;
        font-weight: bold;
        line-height: normal;
        text-align: center;
        letter-spacing: 0em;

        color: #FFFFFF;
      }
    }

    .svg_block {
      position: absolute;
      left: 960rem;
      top: 0;

      img {
        width: 960rem;
        height: 800rem;
      }
    }

    .sec_title {
      position: absolute;
      left: 870rem;
      top: 900rem;
      width: 180rem;
      height: 65rem;
      opacity: 1;

      font-family: Poppins;
      font-size: 43rem;
      font-weight: bold;
      line-height: normal;
      text-align: center;
      letter-spacing: 0em;

      color: #333333;
    }

    #icons {
      img {
        position: absolute;
        width: 60rem;
        height: 60rem;

        &:nth-child(1) {
          left: 300rem;
          top: 1071rem;
        }

        &:nth-child(2) {
          right: 753rem;
          top: 1071rem;
        }

        &:nth-child(3) {
          left: 300rem;
          top: 1275rem;
        }

        &:nth-child(4) {
          right: 753rem;
          top: 1275rem;
        }
      }
    }

    .thd_title {
      position: absolute;
      // left: 870rem;
      left: 50%;
      top: 1519rem;
      // width: 197rem;
      height: 65rem;
      transform: translate(-50%);
      opacity: 1;

      font-family: Poppins;
      font-size: 43rem;
      font-weight: bold;
      line-height: normal;
      text-align: center;
      letter-spacing: 0em;

      color: #333333;
    }

    .thd_svg_block {
      position: absolute;
      left: 240rem;
      top: 1644rem;

      img {
        width: 1440rem;
        height: 440rem;
      }

      .thd_dsp_one {
        position: absolute;
        left: 806rem;
        top: 130rem;
        width: 484rem;
        height: 78rem;
        opacity: 1;

        font-family: Poppins;
        font-size: 26rem;
        font-weight: bold;
        line-height: normal;
        letter-spacing: 0em;

        color: #333333;

      }

      .thd_dsp_two {
        position: absolute;
        left: 805rem;
        top: 228rem;
        width: 448rem;
        height: 78rem;
        opacity: 1;

        font-family: Poppins;
        font-size: 17rem;
        font-weight: normal;
        line-height: normal;
        letter-spacing: 0em;

        color: #5A5A5A;

      }
    }
  }

  .tail {
    position: absolute;
    top: 2144rem;
    width: 100%;
    height: 200rem;
    background-color: #0871F0;

    // border-bottom: 1rem solid #000;
    .title {
      position: absolute;
      left: 301rem;
      top: 50rem;
      width: 887rem;
      height: 96rem;
      opacity: 1;

      font-family: Poppins;
      font-size: 36rem;
      font-weight: bold;
      line-height: 48rem;
      letter-spacing: 0em;

      color: #FFFFFF;

    }

    .reserveButton {

      position: absolute;
      right: 359rem;
      top: 70rem;
      width: 160rem;
      height: 60rem;
      border-radius: 8rem;
      border: 0;
      opacity: 1;

      background: linear-gradient(0deg, #00BFFF, #00BFFF), #080808;

      cursor: pointer;

      span {
        /* Book demo */

        position: absolute;
        left: 34rem;
        top: 18rem;
        width: 93rem;
        height: 24rem;
        opacity: 1;

        font-family: Poppins;
        font-size: 16rem;
        font-weight: bold;
        line-height: normal;
        text-align: center;
        letter-spacing: 0em;

        color: #FFFFFF;
      }
    }
  }

}


footer {
  .foot {
    span {
      position: absolute;
      top: 2420rem;
      height: 21rem;
      opacity: 1;

      font-family: Poppins;
      font-size: 14rem;
      font-weight: 500;
      line-height: normal;
      letter-spacing: 0em;

      color: #999999;
    }

    // .cpr {
    //   left: 240rem;
    // }
    // .ba {
    //   right: 240rem;
    // }
    span {
      &:nth-child(1) {
        left: 240rem;
      }

      &:nth-child(2) {
        right: 240rem;
      }
    }
  }
}