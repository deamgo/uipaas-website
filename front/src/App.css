* {
  padding: 0;
  margin: 0;
}

a {
  text-decoration: none;
}

ul {
  list-style: none;
}

header {
  position: fixed;
  width: 100%;
  z-index: 1001;
}

header .head {
  height: 88px;
  background-color: purple;
  opacity: 0.3;
}

main {
  width: 100%;
}

main .title {
  position: absolute;
  left: 104px;
  top: 194px;
  width: 520px;
  height: 116px;
  opacity: 1;

  font-family: Source Han Sans;
  font-size: 40px;
  font-weight: 900;
  line-height: normal;
  display: flex;
  align-items: center;
  letter-spacing: 0em;

  font-feature-settings: "kern" on;
  color: #6366F1;
}

main .container {
  height: 100vh;
  border-bottom: 1px solid #000;
}

main .tail {
  height: 209px;
  border-bottom: 1px solid #000;
}

footer {
  width: 100%;
  height: 88px
}