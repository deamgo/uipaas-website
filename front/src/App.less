* {
    padding: 0;
    margin: 0;
    font-family: Source Han Sans;
  }
  
  a {
    text-decoration: none;
  }
  
  ul {
    list-style: none;
  }
  
  header {
    position: fixed;
    width: 100%;
    z-index: 1001;
    .head {
        height: 102px;
        background-color: #FFFFFF;
        // opacity: 0.3;
        .logo {
          position: relative;
          left: 104px;
          top: 50px;
        }
    }
  }
  
  main {
    width: 100vw;
    .container {
        --Hg: '';
        --Wd: 100vw;
        height: 100vh;
        border-bottom: 1px solid #000;
        .text_dsp_block {
            position: absolute;
            left: 103px;
            // top: 193px;
            top: calc(100vh * (193/900));
            width: 525px;
            height: 350px;
            .title {

                position: relative;
                left: 0;
                top: 0;
                width: 520px;
                height: 116px;
                opacity: 1;
            
                font-size: 40px;
                font-weight: 900;
                line-height: normal;
                display: flex;
                align-items: center;
                letter-spacing: 0em;
            
                font-feature-settings: "kern" on;
                color: #6366F1;
            }
            .gary {
                position: relative;
                left: 0;
                top: 10px;
                width: 426px;
                height: 43px;
                opacity: 1;
                  
                
                font-size: 30px;
                font-weight: normal;
                line-height: normal;
                text-align: center;
                letter-spacing: 0.1em;
                  
                font-feature-settings: "kern" on;
                color: #767676;
                  
              }
            .mark {
          
                position: relative;
                left: 2px;
                top: 25px;
                width: 204px;
                height: 52px;
                opacity: 1;
                  
                
                font-size: 36px;
                font-weight: 900;
                line-height: normal;
                text-align: center;
                display: flex;
                align-items: center;
                letter-spacing: 0.1em;
                  
                font-feature-settings: "kern" on;
                color: #767676;
                  
            }
            .reserveButton {
                position: relative;
  
                left: 0px;
                top: 53px;
                width: 194px;
                height: 48px;
                border-radius: 5px;
                border: 0px;
                opacity: 1;
                  
                background: #6366F1;
                box-shadow: 0px 3px 8px 0px rgba(0, 0, 0, 0.16);

                cursor: pointer;
                span {
                    height: 35px;
                    opacity: 1;
                        
                    font-family: 思源黑体;
                    font-size: 24px;
                    font-weight: normal;
                    line-height: normal;
                    text-align: center;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    letter-spacing: 0.1em;
                        
                    font-feature-settings: "kern" on;
                    color: #FFFFFF;
	
                }
            }
        }
        .svg_block {
            position: absolute;
            left: 640px;
            top: 163px;
        }
        
    }
    .tail {
        height: 209px;
        border-bottom: 1px solid #000;
    }
  }
  
  footer {
    width: 100%;
    height: 88px
  }