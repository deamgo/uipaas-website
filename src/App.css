* {
  padding: 0;
  margin: 0;
}

a {
  text-decoration: none;
}

ul {
  list-style: none;
}

header {
  position: fixed;
  width: 100%;
  z-index: 1001;
}

header .head {
  height: 88px;
  background-color: purple;
  opacity: 0.3;
}

main {
  width: 100%;
}

.container {
  height: 100vh;
  border-bottom: 1px solid #000;
}

.tail {
  height: 209px;
  border-bottom: 1px solid #000;
}

footer {
  width: 100%;
  height: 88px
}