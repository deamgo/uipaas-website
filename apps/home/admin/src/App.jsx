import { useEffect } from "react"
import { BrowserRouter as Routes, Route } from 'react-router-dom'
import { Navigate } from 'react-router-dom'
import './App.less'
import { getComponyList } from './api/comp_info'

//layout
import StickyHeadTable from './layout/Table'

//svg
import logo from './assets/logov1.svg'

const Admin = () => {

  // const getCompanys = () => {
  //   getComponyList({
  //     pageSize: 10,
  //     pageNum: 1
  //   }).then(res => {
  //     console.log(res);
  //   })
  // }

  // useEffect(() => {
  //   window.addEventListener('resize', resize)
  //   resize()
  // },[])

  // const resize = () => {
  //   let IwD = window.innerWidth
  //   document.documentElement.style.fontSize = IwD/1440 + 'px'
  // }

  const isAuthenticated = localStorage.getItem('jwtToken') !== null

  return (
    <>
      <div className="siderbar">
        <div className="logo-wapper">
          <img src={logo} alt="UIPaaS-Logo" />
        </div>
        <div className="sider-menu">
          <div className="guide">
            <span>内容管理</span>
          </div>
        </div>
      </div>
      <div className="head">
        <div className="cicle"></div>
        <span>管理员</span>
      </div>
      <div className="main">
        <StickyHeadTable />
      </div>
      {/* <button onClick={getCompanys}>BUTTON</button> */}
      {!isAuthenticated && <Navigate to="/logo" />}
    </>
  )
}

export default Admin