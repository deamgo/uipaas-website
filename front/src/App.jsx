//components
import { Paper } from './components/Paper'
import { CollectInfo } from './components/CollectInfo'
//
import { useState, useEffect } from 'react'
//style
import './App.less'
//svg
import logoSvg from './assets/logov1.svg'
import bannerSVG from './assets/ab/banner-pic.svg'
import aboutSvg from './assets/ab/aboutus.svg'
import aiiSvg from './assets/ab/ai-icon.svg'
import modiSvg from './assets/ab/model-icon.svg'
import serviSvg from './assets/ab/server-icon.svg'
import sectiSvg from './assets/ab/section-icon.svg'

let visAble = false

function App() {

  const [btnS, setbtnS] = useState({
    position: 'absolute',
  });

  const [btnVis, setbtnVis] = useState({
    visibility: 'hidden',
  })

  const [tabVis, settabVis] = useState(visAble)
  
  useEffect(() => {
    window.addEventListener('resize', resize)
    resize()
    scrollDetector()
  },[])

  const resize = () => {
    let IwD = window.innerWidth
    document.documentElement.style.fontSize = IwD/1920 + 'px'
  }
  
  const scrollDetector = () => {
    window.addEventListener('scroll', () => {
      let Hbtn = document.querySelector('#hBtn')
      let Hbtn_T = Hbtn.offsetTop
      let Hbtn_H = Hbtn.offsetHeight
      let tTH = Hbtn_H + Hbtn_T
      if (window.scrollY >= tTH) {
        setbtnS({
          position: 'fixed'
        })
        setbtnVis({
          display: 'block'
        })
      } else {
        setbtnS({
          position: 'absolute'
        })
        setbtnVis({
          display: 'none'
        })
      }
    })
  }

  const handleClick = () => {
    visAble = !visAble
    settabVis(visAble)
    document.body.style.overflow = visAble ? 'hidden' : 'auto'
    document.body.style.position = visAble ? 'fixed' : ''
  }

  return (
    <>
      <header  style={btnS}>
        <div 
          className='head'>
            <img className='logo' src={logoSvg} alt="Logo" />
              <button 
                className='reserveButton' 
                style={btnVis}
                onClick={handleClick}>
                  <span>Book demo</span>
              </button>
            {/* <div className="reserve" id="headBtn" style={btnVis}>
            </div> */}
        </div>
      </header>
      <main>
        <div className="container">
          <div className="title">
            <span>
              AI-base, future-oriented, 
              <br />
              drag-and-drop
              <br />
              development platform.
            </span>
          </div>
          <div className="dsp">
            <span>
              The agile delivery tool for software development teams.
              <br/>
              Delivering software promptly with a business-oriented approach.
            </span>
          </div>
            <button
              className='reserveButton' 
              id="hBtn"
              onClick={handleClick}>
                <span>Book demo</span>
            </button>
          {/* <div className="reserve" id="homeBtn">
          </div> */}
          <div className='svg_block'>
              <img src={bannerSVG} alt='feature' />
          </div>
          <div className="sec_title">
            <span>Feature</span>
          </div>
          <div id="icons">
            <img src={aiiSvg} alt="AI base" />
            <img src={modiSvg} alt="Easier use" />
            <img src={serviSvg} alt="deploy" />
            <img src={sectiSvg} alt="highly" />
          </div>
          <Paper 
            title={"AI base"} 
            content={"Creating a New Paradigm for Application Development"} 
            ps={{
              left: '395rem',
              top: '1065rem',
            }}
          />
          <Paper
            title={"Easier-to-use model-driven products"}
            content={"Focus on business, no modeling expertise required"}
            ps={{
              left: '1202rem',
              top: '1065rem',
            }}
          />
          <Paper
            title={"Application-independent deployment"}
            content={"Support applications to run independently from the platform, one-click rapid deployment"}
            ps={{
              left: '395rem',
              top: '1269rem',
            }}
          />
          <Paper
            title={"Highly expandable"}
            content={"Support for custom components, pluggable integration"}
            ps={{
              left: '1202rem',
              top: '1269rem',
            }}
          />
          <div className="thd_title">
            <span>About us</span>
          </div>
          <div className="thd_svg_block">
            <img src={aboutSvg} alt="About" />
            <span className="thd_dsp_one">
              A team of senior technical experts 
              <br />
              in the low-code field
            </span>
            <span className="thd_dsp_two">
              Our Vision is to provide development teams with a 
              <br />
              new development tool that can be used to improve 
              <br />
              the efficiency of delivering</span>
          </div>
        </div>
        <div className="tail">
          <span className="title">
            AI-base, future-oriented,
            <br />
            drag-and-drop  development platform.
          </span>
          <button 
            className="reserveButton"
            onClick={handleClick}>
            <span>Book demo</span>
          </button>
        </div>
      </main>
      <footer>
        <div className="foot">
          <span className="cpr">Copyright © Deamoy Technology</span>
          <span className="ba">浙ICP备2021001545号-4</span>
        </div>
      </footer>
      {tabVis && <CollectInfo onClose={handleClick}/>}
    </>
  ); 
}

export default App;
