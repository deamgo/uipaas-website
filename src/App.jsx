import './App.css'

function App() {

  return (
    <>
      <header>
        <div className='head'></div>
      </header>
      <main>
        <div className='container'></div>
        <div className='container'></div>
        <div className='container'></div>
        <div className='tail'></div>
      </main>
      <footer>
        <div className='foot'></div>
      </footer>
    </>
  )
}

export default App
