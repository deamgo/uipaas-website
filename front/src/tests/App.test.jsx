import React from 'react'
import { render } from '@testing-library/react'
import App from '../App.jsx'
import Err from '../layout/errorPage.jsx'
import PrivacyP from '../layout/Privacy-policy.jsx'
import { CollectInfo } from '../components/CollectInfo/index.jsx'

//router
import { createMemoryRouter, RouterProvider } from 'react-router-dom'


describe('Main', () => {
    it('App correct render', () => {
        const { baseEL } = render(<App />)
        expect(baseEL).toMatchSnapshot()
    })
    it('Privacy-Policy correct render', () => {
        const { baseEL } = render(<PrivacyP />)
        expect(baseEL).toMatchSnapshot()
    })
    it('Err page correct render', () => {
        const routes = [
            {
                path: '/',
                errorElement: <Err />   
            },
        ]
        const router = createMemoryRouter(
            routes,
            {
                initialEntries: ['/eee'],
                initialIndex: 1,
            }
        )
        const { baseEL } = render(<RouterProvider router={router}><App /></RouterProvider>)
        expect(baseEL).toMatchSnapshot()
    })


    test('calls onClick function when clicked', () => {
        const mockOnClick = jest.fn();
        const { asFragment } = render(<CollectInfo/>);
        expect(asFragment()).toMatchSnapshot();
      });
})