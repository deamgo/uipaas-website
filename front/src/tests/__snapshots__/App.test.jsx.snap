// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App correct render 1`] = `undefined`;

exports[`Main Err page correct render 1`] = `undefined`;

exports[`Main Privacy-Policy correct render 1`] = `undefined`;
